### PR TITLE
Reader Cold Start: only count follows in current session

### PR DESCRIPTION
**Pull request by bluefuton**
Presenting the user with a count of their total followed sites _without_ sites they own has proven difficult, so we're trying a different approach.

With this PR, Cold Start only counts the number of Cold Start follows in the current session. Now `/recommendations/start` will encourage you to follow at least one new thing on every visit.

Test live: https://calypso.live/recommendations/start?branch=try/reader/cold-start-follow-count-per-session

### CODE REVIEW

**Comment by bluefuton** [comment]
Todo:
- [ ] fire `recordRecommendationFollow` action on follow

**Comment by bluefuton** [comment]
@martinremy pointed out:

"that all makes sense and is a good solution, but should we also count follows that happen after I click a blog or post title in CS, then follow from the blog or post view?"

So we should fire the action from `ReaderFollowButton` ideally.

**Comment by bluefuton** [comment]
> So we should fire the action from ReaderFollowButton ideally.

We'll need the recommendation to do this though - potentially passed along in the query string to the full post view and site stream.

**Comment by bluefuton** [comment]
Todo: 
- [ ] if recommendation ID is passed to Reader follow button, fire Redux actions
- [ ] remove manual firing of Redux actions in reader/start and pass recommendation ID instead
- [ ] fix passing of recommendation ID via pretty URLs (e.g. /discover), currently lost
- [ ] pick up recommendation ID from query string in site stream, feed stream and full post and supply to Reader follow button

**Comment by bluefuton** [comment]
This does not currently work for recommendations obtained via search :/

File: client/reader/start/card-header.jsx [hunk 1]
```diff
@@ -8,17 +8,18 @@ import FollowButton from 'reader/follow-button';
...
 				<SiteIcon site={ site } size={ 30 } />
 			</a>
 			<div className="reader-start-card__site-info">
-				<a href={ `/read/blogs/${site.ID}` }>
+				<a href={ siteStreamUrl }>
 					<h1 className="reader-start-card__site-title">{ site.title }</h1>
 				</a>
 				<div className="reader-start-card__follower-count">
-					<a href={ `/read/blogs/${site.ID}` }>{ subscribersCount } followers</a>
+					<a href={ siteStreamUrl }>{ subscribersCount } followers</a>
```
Comment (blowery) [inline client/reader/start/card-header.jsx]: yay DRYing up

File: client/reader/start/main.jsx [hunk 2]
```diff
@@ -77,35 +53,32 @@ const Start = React.createClass( {
 	},
 
 	render() {
-		const totalSubscriptions = this.state.totalSubscriptions;
-
-		// Reduce the total subscription count by one for display (exclude the user's own site)
-		const totalSubscriptionsDisplay = totalSubscriptions > 0 ? totalSubscriptions - 1 : 0;
-		const canGraduate = ( this.state.totalSubscriptions > 1 );
+		const followCount = this.props.followCount;
+		const canExit = ( followCount > 0 );
```
Comment (blowery) [inline client/reader/start/main.jsx]: this is so much nicer

File: client/reader/start/main.jsx [hunk 3]
```diff
@@ -77,35 +53,32 @@ const Start = React.createClass( {
...
 				{ /* Following at least one or more sites */ }
-				{ canGraduate && hasRecommendations &&
+				{ canExit && hasRecommendations &&
 					<div className="reader-start__bar is-following">
 						<span className="reader-start__bar-text">
 							{
 								this.translate(
-									'You\'re following %(totalSubscriptionsDisplay)d site.',
-									'You\'re following %(totalSubscriptionsDisplay)d sites.',
+									'You\'re following %(followCount)d new site.',
```
Comment (blowery) [inline client/reader/start/main.jsx]: maybe these should be past tense? "You've followed 1 new site / You've followed n new sites"?

**Comment by blowery** [comment]
Code looks good, testing now

**Comment by blowery** [comment]
:ship: pending the possible phrasing change. @fraying ?

**Pull request merged**